lib/end_of_life.rb

```diff
@@ -59,7 +60,11 @@ def fetch_repositories(options)
 
     def filter_repositories_with_end_of_life(repositories, max_eol_date:)
       with_loading_spinner("Searching for EOL Ruby in repositories...") do
-        repositories.filter { |repo| repo.eol_ruby?(at: max_eol_date) }
+        Sync do
+          repositories
+            .tap { |repos| repos.map { |repo| Async { repo.ruby_version } }.map(&:wait) }
```

Comment:
If you are doing this a lot, you might find https://github.com/socketry/async-await/blob/main/lib/async/await/enumerable.rb#L24-L30 convenient.

Reply:
That is pretty cool, @ioquatix. Could that be included in Async optionally? So one would include `async/enumerable` to get those methods?

Reply:
I think it's better this is a separate gem, as it's a monkey patch.